# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/InternalRequest.h
# Commit: 5c3b5a384f2f
# Full Hash: 5c3b5a384f2fb0ee756d2fe024d2cda4fb643aa0
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-12 21:30:33
# Regressor Bug: 1351231
# File Overlap Count: 1
# Description:
#   Bug 1351231 - Support conversation between InternalRequest and IPCInternalRequest. r=dom-worker-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138813
# ==============================================================================

diff -r d8fa8a006948 -r 5c3b5a384f2f dom/fetch/InternalRequest.h
--- a/dom/fetch/InternalRequest.h	Thu Jan 12 15:09:44 2023 +0000
+++ b/dom/fetch/InternalRequest.h	Thu Jan 12 15:09:44 2023 +0000
@@ -29,6 +29,7 @@
 namespace mozilla {
 
 namespace ipc {
+class PBackgroundChild;
 class PrincipalInfo;
 }  // namespace ipc
 
@@ -97,6 +98,9 @@
 
   explicit InternalRequest(const IPCInternalRequest& aIPCRequest);
 
+  void ToIPCInternalRequest(IPCInternalRequest* aIPCRequest,
+                            mozilla::ipc::PBackgroundChild* aManager);
+
   SafeRefPtr<InternalRequest> Clone();
 
   void GetMethod(nsCString& aMethod) const { aMethod.Assign(mMethod); }